RELEASE/scripts/autoscend/auto_buff.ash

```diff
@@ -99,32 +99,62 @@ boolean buffMaintain(item source, effect buff, int uses, int turns, boolean spec
 	{
 		int numToBuy = uses - item_amount(source);
 		int meatAvailableToBuy = my_meat() - meatReserve();
+		int meatLimit = min(meatAvailableToBuy, get_property("autoBuyPriceLimit").to_int());		
+
 		// attempt to buy from NPC for meat
-		if(npc_price(source) != 0 && meatAvailableToBuy > npc_price(source))
+		if(npc_price(source) != 0)
 		{
-			if(!speculative)
+			if(npc_price(source) < meatLimit)
```

Comment:
on a related note perhaps you would consider some more hardcoded limits for
>8, $item[Star]
>7, $item[line]
like dividing by the amount if limit is above a large value, or dividing by 2 or 3 = the amount you can get with one adventure, or maybe taking into account the property "valueOfAdventure"
because unlike most pulls these items will be trivially acquired with adventuring and I think users would not expect to buy up to their price limit value for each of 8 stars